ee/packages/media-calls/src/server/CallDirector.ts

```diff
@@ -0,0 +1,432 @@
+import type { IMediaCall, IMediaCallNegotiation, MediaCallContact, MediaCallSignedActor, ServerActor } from '@rocket.chat/core-typings';
+import type { CallHangupReason, CallRole } from '@rocket.chat/media-signaling';
+import type { InsertionModel } from '@rocket.chat/model-typings';
+import { MediaCallNegotiations, MediaCalls } from '@rocket.chat/models';
+
+import { getCastDirector } from './injection';
+import type { IMediaCallAgent } from '../definition/IMediaCallAgent';
+import type { IMediaCallCastDirector } from '../definition/IMediaCallCastDirector';
+import type { InternalCallParams, MediaCallHeader } from '../definition/common';
+import { logger } from '../logger';
+
+const EXPIRATION_TIME = 120000;
+const EXPIRATION_CHECK_TIMEOUT = EXPIRATION_TIME + 1000;
+
+export type CreateCallParams = InternalCallParams & {
+	callerAgent: IMediaCallAgent;
+	calleeAgent: IMediaCallAgent;
+
+	webrtcOffer?: RTCSessionDescriptionInit;
+};
+
+// expiration checks by call id
+const scheduledExpirationChecks = new Map<string, ReturnType<typeof setTimeout>>();
+
+export class MediaCallDirector {
```

Comment:
[nitpick] Consider refactoring it to not use a class as a namespace for functions.

Reply:
Will do it on another PR.